### ISSUE TITLE
Ryujinx won't start on Windows 7 after .NET 6 migration

### ISSUE THREAD

**Issue by pkt01** [Ryujinx/Ryujinx#2889]
### What's the issue you encountered?

When launching Ryujinx on Windows 7, the system error message *"The program can't start because api-ms-win-core-winrt-l1-1-0.dll is missing"* pops up. Windows 7 apparently lacks certain API dlls added in 8 and 10.

### How can the issue be reproduced?

Start Ryujinx 1.0.7111 on Windows 7.

### Log file

N/A.

### Environment?

- Ryujinx version: 1.0.7111</br>
- Game version: N/A.</br>
- System Specs:
  - OS: *Windows 7 x64*
  - CPU: *i7-4770*
  - GPU: *NVIDIA GTX 1050 Ti*
  - RAM: *8GB*
- Applied Mods : N/A.

### Additional context?

.NET 6 desktop runtime and all VC redists are installed.

The **workaround** (besides upgrading the OS) is to [download](https://www.nuget.org/packages/runtime.win7-x64.Microsoft.NETCore.Windows.ApiSets) the missing API dlls package and then extract *api-ms-win-core-winrt-l1-1-0.dll* (found in \runtimes\win7-x64\native) to the same directory as the Ryujinx executable. Emulator starts and runs fine afterwards.

It is not a Ryujinx problem per se, more of a side effect of .NET 6 lacking full support for Windows 7. The problem was discussed [here](https://github.com/dotnet/runtime/issues/55514).

**Comment by poudink** [comment]
I'd like to add that Ryujinx appears to delete (or to add the .ryuold extension to) the DLL when auto-updating, so the emulator restart fails and it has to be re-added.

**Comment by Thog** [comment]
Windows 7 is out of support from Microsoft since **January 14, 2020**.
The more time pass, the more difficult it is to target it as it's entirely unsupported those days (even the official CRT will try to load missing dependencies for example)

I do agree that this is a major regression however, I don't see the point of using a bleeding edge emulator on such operating system considering even [NVIDIA dropped support of their drivers on it since October 2021](https://nvidia.custhelp.com/app/answers/detail/a_id/5201/related/1).

**Comment by pkt01** [comment]
Agreed. The idea was to merely offer a "temporary fix" to people who may be stuck with Windows 7.

**Comment by Nukepayload2** [comment]
`api-ms-win-core-winrt-l1-1-0.dll` can be found in [runtime.win7-x64.microsoft.netcore.windows.apisets.1.0.1.nupkg](https://www.nuget.org/api/v2/package/runtime.win7-x64.Microsoft.NETCore.Windows.ApiSets/1.0.1)

![image](https://user-images.githubusercontent.com/10798596/144859704-dc2ecc58-e31e-41e4-b066-5997d3b0bc62.png)

My customers are also facing this problem. I suggested that they should decompress `runtime.win7-x64.microsoft.netcore.windows.apisets.1.0.1.nupkg!\runtimes\win7-x64\native\*.*` to `%PATH%` before running my .NET 6 apps.

**Comment by Shahil-Ayato** [comment]
IMHO, Ever since the .net 5 update windows 7 support has been hit or miss. I understand trying our best to support legacy OS’s but windows 7 is slowly becoming unsupported by many other projects and software. I think we should close this issue and/or drop support for windows 7 all together because it’s really on the user for being on an OS that’s so old and insecure, and it’s not as if updating the OS is very difficult anyways. If a users hardware somehow can’t support Windows 10 anyways then more than likely it is much too weak for Ryujinx.

**Comment by GamerzHell9137** [comment]
> IMHO, Ever since the .net 5 update windows 7 support has been hit or miss. I understand trying our best to support legacy OS’s but windows 7 is slowly becoming unsupported by many other projects and software. I think we should close this issue and/or drop support for windows 7 all together because it’s really on the user for being on an OS that’s so old and insecure, and it’s not as if updating the OS is very difficult anyways. If a users hardware somehow can’t support Windows 10 anyways then more than likely it is much too weak for Ryujinx.

iirc even Avalonia's theme needs to be changed for it to work on W7. Its a hassle at this point + MS is not supporting it.

**Comment by okt04175** [comment]
.net 7 maybe will resolve this issue.

[agocke](https://github.com/agocke) added this to the [7.0.0](https://github.com/dotnet/runtime/milestone/82) milestone [on 29 Jul 2021](https://github.com/dotnet/runtime/issues/55514#event-5082885848)

[](https://github.com/apps/msftbot)@[msftbot](https://github.com/apps/msftbot) msftbot bot locked as resolved and limited conversation to collaborators [on 17 Mar](https://github.com/dotnet/runtime/issues/55514#event-6254916118)

**Comment by jduncanator** [comment]
Regardless of whether .NET 7 will support Windows 7 or not, the get the most out of Ryujinx you will need a fully upto date install of Windows 10.

With some changes that are due to be merged to the memory manager shortly, you will be forced onto a slow path for memory accesses on anything but the latest Windows 10/11 OSes, which will lead to a significantly impaired experience on Ryujinx. In my mind, it makes zero sense to remain on Windows 7.

**Comment by Thog** [comment]
See https://github.com/Ryujinx/Ryujinx/issues/3298.

**Comment by sauradipdebnath** [comment]
No making useless issues
